tests/assistant_spec.js

```diff
@@ -831,6 +868,157 @@ describe('TranslationAssistant (math-translate)', function() {
     });
 });
 
+describe('TranslationAssistant (maybe-math-translate)', function() {
+
+    it('should translate math according to a template', function() {
+        // Lang 'id' is in both MAYBE_TIMES_AS_CDOT and MAYBE_DIV_AS_COLON
+        const lang = 'id';
+
+        const allItems = [{
+            englishStr: 'simplify $2 \\times 2 \\div 2$',
+            translatedStr: 'simplifyz $2 \\cdot 2 \\mathbin{:} 2$',
+        }];
+        const itemsToTranslate = [{
+            englishStr: 'simplify $4 \\div 1 \\times 5$',
+            translatedStr: '',
+        }];
+        // The translated template contained mathbin and cdot, so we will
+        // translate accordingly
+        const translatedStrs = ['simplifyz $4 \\mathbin{:} 1 \\cdot 5$'];
+
+        assertSuggestions(allItems, itemsToTranslate, translatedStrs, lang);
+
+        // Here the translator translated div as mathbin,
+        // but left times as multiplication operator.
+        const allItems2 = [{
+            englishStr: 'simplify $2 \\times 2 \\div 2$',
+            translatedStr: 'simplifyz $2 \\times 2 \\mathbin{:} 2$',
+        }];
+
+        // Notice times instead of cdot
+        const translatedStrs2 = ['simplifyz $4 \\mathbin{:} 1 \\times 5$'];
+
+        assertSuggestions(allItems2, itemsToTranslate, translatedStrs2, lang);
+    });
+
+    it('should handle maybeTranslateMath together with \\text', function() {
+        const lang = 'id';
+
+        const allItems = [{
+            englishStr: 'simplify $2\\text{to} 2\\times\\div2 \\textbf{equal}$',
+            translatedStr: 'simplifyz $2\\text{t} 2\\cdot\\div2 \\textbf{eq}$',
+        }];
+        const itemsToTranslate = [{
+            englishStr: 'simplify $4\\text{to} 4\\times\\div1 \\textbf{equal}$',
+            translatedStr: '',
+        }];
+        const translatedStrs =
+            ['simplifyz $4\\text{t} 4\\cdot\\div1 \\textbf{eq}$'];
+
+        assertSuggestions(allItems, itemsToTranslate, translatedStrs, lang);
+    });
+
+    it('should return the same math without a template', function() {
```

Comment:
I wonder if `it.each` is possible:

```
it.each(
    MATH_RULES_LOCALES.MAYBE_TIMES_AS_CDOT
)("should return the same math without a template for %s", function(lang) {

});
```

Reply:
Apparently not. :-( We could iterate over locales inside the callback in the same way we do in `math_translater_spec`, but I'll keep it as is for now. I'll try to clean up the tests in some future PR. A lot of them could be made clearer if the lang was chosen more explicitly (e.g. via asserting that lang is in certain rules as you suggested above).